### PR TITLE
Swift 4

### PR DESCRIPTION
**Pull request by dfed**
Built on top of #49.

@mthole or @pwesten, mind testing this in Cash/Caviar before this branch goes to master? I no longer work on an app that uses Aardvark 😞.

I'm hoping the `@objc` changes are fine, since we were already explicitly calling out which swift methods were objc-available.

### CODE REVIEW

**Comment by manolosavi** [comment]
I'll test on Caviar, I'm having some unrelated build issues but I'll let you know when I figure it out. From looking at the changes it looks to me like it should be okay tho.
And fwiw we won't be switching to Swift 4 yet, waiting on CI. The changes I made on friday work were just to start preparing for that, which is why I only changed that warning and not upgraded the whole thing to Xcode 9.

**Comment by dfed** [comment]
Ahhh got it. The Xcode 9 changes should be backwards compatible with Xcode 8. Might as well kill two birds with one stone. I believe that the changes from #49 can be just a bugfix version bump.

I'll hold off on Swift 4 for now. Though when I do merge it it'll be a major version bump, so shouldn't break anything :)

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 language: objective-c
-osx_image: xcode8
+osx_image: xcode9
```
Comment (justinseanmartin) [inline .travis.yml]: If you want to ensure that everything stays backwards compatible, could consider keeping both versions running on Travis.

Example from KIF: https://github.com/kif-framework/KIF/blob/master/.travis.yml

**Comment by justinseanmartin** [comment]
LGTM, just the one suggestion if we're worried about breaking backwards compatibility.

File: .travis.yml (hunk 1 above)
Comment (dfed) [reply .travis.yml]: TIL that was possible!

Though it's worth noting the move to Swift 4 will break Xcode 8. This will need to be a major version bump.

**Comment by manolosavi** [comment]
Looks good here as well!

**Comment by manolosavi** [comment]
Just noticed an issue, it's missing an `@objc` for `public static func addDefaultBugReportingGestureWithEmailBugReporter(withRecipient emailAddress: EmailAddress) -> ARKEmailBugReporter`. The `@objc` is now required in Swift 4.

**Comment by manolosavi** [comment]
There's a few places that use `@nonobjc` which isn't necessary anymore, but it seems like it's good to merge now?
cc/ @mthole

**Comment by dfed** [comment]
@mthole feel free to merge / tag / publish. Since I'm not actively developing an app that's using Aardvark (hoping to change that in the medium-term), it may make sense to have someone who is manage lifecycle.

**Comment by dfed** [comment]
cc @gfontenot @NickEntin – y'all may want to merge/publish this sooner rather than later

**Comment by dfed** [comment]
Closing this in favor of #71.

**Pull request closed without merging**